Change owner of GOPATH to vagrant

rdimartino opened the pull request:
Currently, the directory created for the `GOPATH` is owned by root. This causes permission errors if you try to `go get` something.

I haven't really tested this, but I think this will create the directory with vagrant as the owner.

Commit:
change owner of GOPATH to vagrant

Commit:
readd -p flag

llimllib commented:
hey sorry I totally missed this because github's notifications system is a mess!

Can you move the `mkdir -p ~/go` line into the `user_provision.sh` file instead of running it in the `provision.sh` script? That should be clearer.

Commit:
move GOPATH creation to user_provision.sh

rdimartino commented:
👍 

@llimllib I moved that line to `user_provision.sh` (I didn't even see that before). I also explicitly set `GOPATH` and added `$GOPATH/bin` to `PATH`. Now you can run `go get ...` without an error and easily use the binary installed `~/go/bin`.
